x/staking/querier/querier.go

```diff
@@ -128,14 +129,47 @@ func NewQueryRedelegationParams(delegatorAddr sdk.AccAddress, srcValidatorAddr s
 	}
 }
 
-func queryValidators(ctx sdk.Context, cdc *codec.Codec, k keep.Keeper) (res []byte, err sdk.Error) {
+func queryValidators(ctx sdk.Context, cdc *codec.Codec, req abci.RequestQuery, k keep.Keeper) ([]byte, sdk.Error) {
+	var params QueryValidatorsParams
+
+	err := cdc.UnmarshalJSON(req.Data, &params)
+	if err != nil {
+		return nil, sdk.ErrInternal(fmt.Sprintf("failed to parse params: %s", err))
+	}
+
 	stakingParams := k.GetParams(ctx)
-	validators := k.GetValidators(ctx, stakingParams.MaxValidators)
+	if params.Limit == 0 {
+		params.Limit = int(stakingParams.MaxValidators)
```

Comment:
So `AllValidators` can be greater than the `MaxValidators` (as it contains bonded and non-bonded validators) - maybe we want to set the page limit to 100 by default but it seems kinda arbitrary. It would be my preference to simply error at a page limit of 0. Can we set default page limits elsewhere in rest?

Reply:
So the REST utils actually does have a default limit -- 30. This is why I manually override it to 0. I wanted to keep this endpoint compatible with it's most intended use -- give me all bonded validators. So clients may omit sending `page` and `limit` and get the expected results.

I don't think we should error when no page or limit is provided.

Reply:
agree here

Reply:
👍